src/modules/petrology/FeatureDetail.js

```diff
@@ -0,0 +1,269 @@
+import React, {useRef, useState} from 'react';
+import {Alert, ScrollView, TextInput, View} from 'react-native';
+
+import {Field, Formik} from 'formik';
+import {Button, ListItem} from 'react-native-elements';
+import MultiSelect from 'react-native-multiple-select';
+import {useDispatch, useSelector} from 'react-redux';
+
+import commonStyles from '../../shared/common.styles';
+import {isEmpty, toTitleCase} from '../../shared/Helpers';
+import * as themes from '../../shared/styles.constants';
+import FlatListItemSeparator from '../../shared/ui/FlatListItemSeparator';
+import SaveAndCloseButton from '../../shared/ui/SaveAndCloseButtons';
+import {formStyles, TextInputField, useFormHook} from '../form';
+import {DEFAULT_GEOLOGIC_TYPES} from '../project/project.constants';
+import {addedCustomFeatureTypes} from '../project/projects.slice';
+import {editedSpotProperties} from '../spots/spots.slice';
+
+const FeatureDetail = (props) => {
+  const dispatch = useDispatch();
+  const [useForm] = useFormHook();
+  const spot = useSelector(state => state.spot.selectedSpot);
+  const projectFeatures = useSelector(state => state.project.project.other_features);
+
+  const selectedFeature = props.selectedFeature;
+  const customFeatureTypes = projectFeatures.filter(feature => !DEFAULT_GEOLOGIC_TYPES.includes(feature.name));
+  let label = useState(selectedFeature.label === undefined ? '' : selectedFeature.label);
+  let name = useState(selectedFeature.name === undefined ? '' : selectedFeature.name);
+  const [typeEnum, setTypeEnum] = useState('');
+  let [type, setType]
+    = useState(selectedFeature.type === undefined ? '' : selectedFeature.type);
+  let [otherType, setOtherType] = useState('');
+  let description = useState(selectedFeature.description === undefined ? '' : selectedFeature.description);
+  const featureForm = useRef(null);
```

Comment:
Change to `formRef` to match all other pages where we have a `useRef` for a form.